### PR TITLE
Port to github actions

### PR DESCRIPTION
**Pull request by vatanaksoytezer**
- Add pre-commit
- Deploy to gh-pages on push
- Remove travis and port build to Github Actions

### CODE REVIEW

File: .github/workflows/industrial_ci_action.yml
Comment (tylerjw) [inline .github/workflows/industrial_ci_action.yml:4]: please rename all these files to have the `.yaml` file extension instead of `.yml`.  Also make this file just `ci.yaml` and make sure you update the readme to point to the new filenames.

File: .github/workflows/industrial_ci_action.yml
Comment (tylerjw) [inline .github/workflows/industrial_ci_action.yml:37]: quotes around this line (and the several below it) are unnecessary

File: .github/workflows/industrial_ci_action.yml
Comment (tylerjw) [inline .github/workflows/industrial_ci_action.yml:39]: here

File: .github/workflows/industrial_ci_action.yml
Comment (tylerjw) [inline .github/workflows/industrial_ci_action.yml:40]: here

File: .github/workflows/industrial_ci_action.yml
Comment (tylerjw) [inline .github/workflows/industrial_ci_action.yml:42]: here too

**Review by tylerjw — COMMENTED** [review]
A couple minor nits.  Also, please rebase this into 2 commits:

* Migrate to GitHub Actions
* Fix formatting errors

**Comment by vatanaksoytezer** [comment]
@tylerjw Thanks for the review, updated with your recommendations.

File: .github/workflows/deploy_website.yaml
Comment (davetcoleman) [inline .github/workflows/deploy_website.yaml:1]: @nbbrooks setup the original deployment scripts for moveit_tutorials, in case you need another reviewer. however he's on vacation this week.

File: README.md
Comment (davetcoleman) [inline README.md:13]:
```suggestion
This repository is currently built automatically by two systems. Github Actions builds the documentation for Noetic and Melodic (because they are the same tutorials),
and ROS Build Farm builds the documentation for older versions:
```
Is this statement correct? Or have we configured GActions to build mutiple versions? I'm only aware of the former

File: README.md
Comment (davetcoleman) [inline README.md:18]: Here we state that ROS Build farm builds Melodic... this conflicts with earlier statement right?

**Review by davetcoleman — COMMENTED** [review]
agreed the formatting should be a different PR than the GActions configuration

File: .github/workflows/deploy_website.yaml
Comment (vatanaksoytezer) [reply .github/workflows/deploy_website.yaml:1]: I would be happy to get another review, for now I've tested in my own gh pages and deployed without any problems.

File: README.md
Comment (vatanaksoytezer) [reply README.md:18]: Buildfarm builds for Melodic + Kinetic, GHA builds for Noeitc + Melodic. I will clarify that

File: README.md
Comment (vatanaksoytezer) [reply README.md:13]:
```
jobs:
  default:
    strategy:
      matrix:
        env:
          - IMAGE: master-source
            CCOV: false
            CATKIN_LINT: true
          - IMAGE: noetic-source
            CCOV: false
            CATKIN_LINT: true
```

Here we specify that the tutorials would be built using melodic and noetic containers seperately. It essentially builds the master branch with two different distros.

**Comment by v4hn** [comment]
Ping @vatanaksoytezer 

I believe you said [in the last maintainer meeting](https://discourse.ros.org/t/moveit-manipulation-wg-meeting-may-27th/20431/3#github-actions-migration-8) that most things have been ported, but the tutorials (as a pretty crucial part) have not.

**Comment by vatanaksoytezer** [comment]
> I believe you said [in the last maintainer meeting](https://discourse.ros.org/t/moveit-manipulation-wg-meeting-may-27th/20431/3#github-actions-migration-8) that most things have been ported, but the tutorials (as a pretty crucial part) have not.

Yep, you are right. I also mentioned that this is waiting for review for some time :(. Would you mind reviewing this for me?

**Comment by rhaschke** [comment]
I will have a look right now.

**Comment by vatanaksoytezer** [comment]
> I will have a look right now.

Thanks a ton!

**Review by rhaschke — CHANGES_REQUESTED** [review]
Please fixup the README file: Both, kinetic and melodic tutorials are built on the build farm. Only the noetic version is built here.

The badges are too verbose (remove GHA and other text) and format them as "title: badge" please.
image

**Comment by vatanaksoytezer** [comment]
> Please fixup the README file: Both, kinetic and melodic tutorials are built on the build farm. Only the noetic version is built here.
> 
> The badges are too verbose (remove GHA and other text) and format them as "title: badge" please.

Sure I'll fix it tonight.

**Comment by vatanaksoytezer** [comment]
@rhaschke Thanks a lot again for reviewing and fixing stuff. This should be ready to merge. One thing to be careful about is to check website after merging to see if deployment is successful to gh pages. I've already tested this in my own github pages, but just to be sure.

**Review by rhaschke — APPROVED** [review]

**Pull request merged**
